### ISSUE TITLE
Remove peru and use plain old bash

### ISSUE THREAD

**Issue by talha131** [Pelican-Elegant/elegant#358]
@iranzo You added peru support to the project. Our CI is dependent on it.

When using it locally I noticed that it was very slow. Looking into it, it turned out, it fetches the repository completely. 

If it used `--depth 1` option and do shallow fetches for the submodules, its speed would have been significantly faster. I don't know if it uses `--jobs` option or not.

I looked into peru repository and discovered they are never going to add `--depth` option.

What I ended up doing was adding these lines to my `Makefile`. See

https://github.com/talha131/onCrashReboot/blob/master/Makefile#L75-L86

The result was significantly faster.

It can be modified to do a pull instead of clone if repository is already cloned. But for our CI purpose current version is enough.

Question is should we move from peru to this simple bash script. Advantage is the speed. I suggest you to try both peru and my make file locally and see the time difference.

If you see the travis logs, it spends a significant time in doing `peru sync`. Besides local development, we need faster CI too.

Bottom line is peru did made our job easier but it has also become a bottleneck.

Let me know what you think as you have lead the efforts of CI and peru integration.

**Comment by iranzo** [comment]
Hi
I started using Peru because it allows me to fetch specific branches or
even files (or folders within repos) and just put them in the final folder
structure.

When using locally it's a lot faster as it caches the data, on CI it's
slower as 'per-CI requirements', each environment is destroyed on
completion, so a new test-from-scratch is started.

In my case, also just leaves the plugins folders with the ones I do use
(not all), so when replicating data across computers it just leaves the
minimum set.

I don't mind changing it, to whatever suits more the project,so go ahead!
Pablo


-- 

Pablo Iranzo Gómez

GnuPG: 0x5BD8E1E4
Web: https://iranzo.github.io  ||  https://citellus.org
